File: eecs168/2020fall/TR/2020.11.17.classes-part-4/Account/Account.cpp
//Account.cpp
#include "Account.h"

Account::Account(double startingBalance)
{
	if(startingBalance > 0)
	{
		balance = startingBalance;
	}
	else
	{
		balance = 0;
	}
}

bool Account::deposit(double amount)
{
	if( amount > 0 )
	{	
		balance = balance + amount;
		return (true);
	}
	else
	{
		return (false);
	}	
}

bool Account::withdraw(double amount) 
{
	if( amount <= balance && amount > 0 )
	{
		balance = balance - amount;
		return (true);
	}
	else
	{
		return (false);
	}	
}

double Account::checkBalance() const
{
	//I shouldn't be changing the balance
	return(balance);
}

bool Account::transfer(Account& destination, double amount)
{
	if(amount > 0 && amount <= balance)
	{
		// destination.deposit(amount);
		
		//Accessing private member is allowed
		//because we're in the Account class
		destination.balance += amount;
		balance = balance - amount;
		return (true);
	}
	else
	{
		return (false);
	}
}
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	
	

File: eecs168/2024spring/TR/2024.04.11.classes-part-4/account.py
#account.py

class Account:
    def __init__(self, start_balance):
        self._balance = start_balance
        
    def __repr__(self):
        output_str = f'Account({self._balance})'
        return output_str
    
    def __str__(self):
        output_str = 'Account with balance: '
        output_str += str(self._balance)
        return output_str
    
    def deposit(self, amount):
        if amount > 0:
            self._balance += amount
        else:
            pass

    def withdraw(self, amount):
        #define this
        pass

    def transfer(self, other, amount):
        #define this
        pass

    def check_balance(self):
        return self._balance

    def __gt__(self, other):
        if self._balance > other._balance:
            return True
        else:
            return False

    def __eq__(self, other):
        if self._balance == other._balance:
            return True
        else:
            return False







    
